pagerduty/resource_pagerduty_event_orchestration_path_router.go

```diff
@@ -0,0 +1,394 @@
+	}
+
+	return orchPath
+}
+
+func expandOrchestrationPathParent(v interface{}) *pagerduty.EventOrchestrationPathReference {
+	var parent *pagerduty.EventOrchestrationPathReference
+	p := v.([]interface{})[0].(map[string]interface{})
+	parent = &pagerduty.EventOrchestrationPathReference{
+		ID:   p["id"].(string),
+		Type: p["type"].(string),
+		Self: p["self"].(string),
+	}
+
+	return parent
+}
+
+func expandSets(v interface{}) []*pagerduty.EventOrchestrationPathSet {
+	var sets []*pagerduty.EventOrchestrationPathSet
+
+	for _, set := range v.([]interface{}) {
+		s := set.(map[string]interface{})
+
+		orchPathSet := &pagerduty.EventOrchestrationPathSet{
+			ID:    s["id"].(string),
+			Rules: expandRules(s["rules"].(interface{})),
+		}
+
+		sets = append(sets, orchPathSet)
+	}
+
+	return sets
+}
+
+func expandRules(v interface{}) []*pagerduty.EventOrchestrationPathRule {
+	items := v.([]interface{})
+	rules := []*pagerduty.EventOrchestrationPathRule{}
+
+	for _, rule := range items {
+		r := rule.(map[string]interface{})
+
+		ruleInSet := &pagerduty.EventOrchestrationPathRule{
+			ID:         r["id"].(string),
+			Label:      r["label"].(string),
+			Disabled:   r["disabled"].(bool),
+			Conditions: expandRouterConditions(r["conditions"].(interface{})),
+			Actions:    expandRouterActions(r["actions"].([]interface{})),
+		}
+
+		rules = append(rules, ruleInSet)
+	}
+	return rules
+}
+
+func expandRouterConditions(v interface{}) []*pagerduty.EventOrchestrationPathRuleCondition {
+	var conditions []*pagerduty.EventOrchestrationPathRuleCondition
+
+	for _, cond := range v.([]interface{}) {
+		c := cond.(map[string]interface{})
+
+		cx := &pagerduty.EventOrchestrationPathRuleCondition{
+			Expression: c["expression"].(string),
+		}
+
+		conditions = append(conditions, cx)
+	}
+
+	return conditions
+}
+
+func expandRouterActions(v interface{}) *pagerduty.EventOrchestrationPathRuleActions {
+	var actions = new(pagerduty.EventOrchestrationPathRuleActions)
+	for _, ai := range v.([]interface{}) {
+		am := ai.(map[string]interface{})
+		actions.RouteTo = am["route_to"].(string)
+	}
+
+	return actions
+}
+
+func expandCatchAll(v interface{}) *pagerduty.EventOrchestrationPathCatchAll {
+	var catchAll = new(pagerduty.EventOrchestrationPathCatchAll)
+
+	for _, ca := range v.([]interface{}) {
+		am := ca.(map[string]interface{})
+		catchAll.Actions = expandRouterActions(am["actions"].(interface{}))
+	}
+
+	return catchAll
+}
+
+func flattenSets(orchPathSets []*pagerduty.EventOrchestrationPathSet) []interface{} {
+	var flattenedSets []interface{}
+	for _, set := range orchPathSets {
+		flattenedSet := map[string]interface{}{
+			"id":    set.ID,
+			"rules": flattenRules(set.Rules),
+		}
+		flattenedSets = append(flattenedSets, flattenedSet)
+	}
+	return flattenedSets
+}
+
+func flattenRules(rules []*pagerduty.EventOrchestrationPathRule) []interface{} {
+	var flattenedRules []interface{}
+
+	if rules != nil {
```

Comment:
rules can be nil (brand new router or a router where all rules are intended to be deleted). we expand on conditions actions only if rules are not empty otherwise return empty list as is.